Limber/MIXMockServer.cpp

```diff
@@ -0,0 +1,263 @@
+                        SWIFT_LOG(debug) << "Request: Channel Leave - " << *channelJID << std::endl;
+                        if (mixChannelRegistry_->hasMIXChannel(*channelJID)) {
+
+                            session->sendElement(IQ::createResult(iq->getFrom(), iq->getTo(), iq->getID(), std::make_shared<MIXLeave>()));
+
+                            auto i = rosterMap_.find(iq->getFrom().toBare());
+                            if (i != rosterMap_.end()) {
+                                auto roster = i->second;
+                                roster->removeContact(*channelJID);
+
+                                auto rosterUpdatePayload = std::make_shared<RosterPayload>();
+                                auto itemPayload = RosterItemPayload(*channelJID, channelJID->getNode(), RosterItemPayload::Subscription::Remove);
+                                itemPayload.setMIXChannel(true);
+                                rosterUpdatePayload->addItem(itemPayload);
+
+                                session->sendElement(IQ::createRequest(IQ::Set, iq->getFrom(), iq->getID(), rosterUpdatePayload));
+                            } else {
+                                SWIFT_LOG(debug) << "Initial roster not requested by client." <<std::endl;
+                            }
+                        } else {
+                            SWIFT_LOG(debug) << "Channel " << *channelJID << "not hosted on domain." <<std::endl;
+                        }
+                    } else {
+                        session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+                    }
+                }
+                return;
+            }
+
+            //If request comes to particular channel supported by service.
+            if (stanza->getTo().isValid() && mixChannelRegistry_->hasMIXChannel(stanza->getTo())) {
+                if (iq->getPayload<DiscoItems>()->getNode() == "mix") {
+                    SWIFT_LOG(debug) << "Query: Channel Nodes Supported" << std::endl;
+                    auto responsePayload = std::make_shared<DiscoItems>();
+                    responsePayload->addItem(DiscoItems::Item(MIX::ParticipantsNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::MessagesNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::PresenceNode, JID(iq->getTo())));
+                    responsePayload->addItem(DiscoItems::Item(MIX::JIDMapNode, JID(iq->getTo())));
+                    session->sendElement(IQ::createResult(iq->getFrom(), iq->getTo(), iq->getID(), responsePayload));
+                }
+                else {
+                    session->sendElement(IQ::createError(iq->getFrom(), iq->getID(), ErrorPayload::BadRequest, ErrorPayload::Cancel));
+                }
+                return;
+            }
+        }
```

Comment:
This method is quite long. It probably makes sense to split it up in parts. Like `handleElementReceivedForAccount`, `handleElementReceivedForMIXService`, `handleElementReceivedForMIXChannel`, etc.